### PR TITLE
WIP apollo server v4

### PR DESCRIPTION
**Pull request by sschneider-ihre-pvs**
## PR Checklist
Please check if your PR fulfills the following requirements:

- [ ] The commit message follows our guidelines: https://github.com/nestjs/nest/blob/master/CONTRIBUTING.md
- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)

## PR Type
What kind of change does this PR introduce?

- [ ] Bugfix
- [ ] Feature
- [ ] Code style update (formatting, local variables)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] CI related changes
- [ ] Other... Please describe:

## What is the current behavior?

Issue Number: N/A

## Does this PR introduce a breaking change?
- [ ] Yes
- [ ] No

### CODE REVIEW

**Comment by sschneider-ihre-pvs** [comment]
This is just a quick overview of what I figured out so far

File: packages/apollo/lib/drivers/apollo-errors.driver.ts [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+
+    if (!this.name) {
+      Object.defineProperty(this, 'name', { value: 'ApolloError' });
+    }
+
+    this.extensions = { ...extensions, code };
+  }
+}
+
+export class UserInputError extends ApolloError {
```
Comment (kamilmysliwiec) [inline packages/apollo/lib/drivers/apollo-errors.driver.ts:25]: Why do we need these classes to be included in this package?

File: packages/apollo/package.json [hunk 2]
```diff
@@ -38,6 +41,9 @@
     "apollo-server-express": "3.11.1",
     "apollo-server-fastify": "3.11.1",
     "apollo-server-plugin-response-cache": "3.8.1",
+    "body-parser": "^1.20.1",
+    "cors": "^2.8.5",
+    "express": "^4.18.2",
```
Comment (Maxwell2022) [inline packages/apollo/package.json:46]: I think all these should be peer dependencies base on what server your want to use (express or fastify)

File: packages/apollo/lib/drivers/apollo-errors.driver.ts (hunk 1 above)
Comment (sschneider-ihre-pvs) [reply packages/apollo/lib/drivers/apollo-errors.driver.ts:25]: If you want to keep backwards compatibility only.

**Comment by LeoKaynan** [comment]
I'm looking forward to. ><

**Comment by hinogi** [comment]
Maybe I should continue the work on this rough draft

**Comment by cabrinoob** [comment]
Is this PR mean that @nest/graphql module will provide a @cacheControl directive ?

**Comment by hinogi** [comment]
maybe close this pr in favor of @luas10c

**Comment by kamilmysliwiec** [comment]
Let's track this here https://github.com/nestjs/graphql/pull/2631

**Pull request closed without merging**